src/Tests/Microsoft.NET.Sdk.WorkloadManifestReader.Tests/SdkDirectoryWorkloadManifestProviderTests.cs

```diff
@@ -9,25 +9,37 @@
 using Microsoft.NET.TestFramework;
 using System.Linq;
 using Microsoft.NET.Sdk.WorkloadManifestReader;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
 
 namespace ManifestReaderTests
 {
+    //  Don't run in parallel with any other tests which set DOTNETSDK_WORKLOAD_MANIFEST_ROOTS in-process
+    //  (they also need to be annotated with this Collection)
+    [Collection("DOTNETSDK_WORKLOAD_MANIFEST_ROOTS")]
```

Comment:
If we need to override DOTNETSDK_WORKLOAD_MANIFEST_ROOTS in process, that means we should be able to mock it out in object level, no? In the case we cannot mock it out, it should be in a separate process, and in this case they are isolated by different processes already.

Reply:
Good suggestion about mocking out the environment, I've fixed this.